src/uwtools/rocoto.py

```diff
@@ -25,20 +35,117 @@ def _add_jobname(tree: dict) -> None:
             _add_jobname(subtree)
 
 
-# Public functions
-def write_rocoto_xml(input_yaml: str, input_template: str, rendered_output: str) -> None:
+def _add_tasks(
```

Comment:
This name is misleading. It doesn't add tasks to the YAML. Since this calls `_add_jobname` for all tasks, perhaps it could be `_add_jobnames` or `_add_jobname_to_tasks`?

Also, the function isn't returning the updated values, so I'm not sure these jobnames are going to be present when you write the xml.

Reply:
Ah, thank you!